src/CommonDBTM.php

```diff
@@ -1185,6 +1211,25 @@ protected function restoreInput(array $default = [])
         return $default;
     }
 
+    final public function callFormAction(string $form_action, array $input): mixed
+    {
+        // POST action execution
+        return match ($form_action) {
+            'add' => $this->add($input),
+            'delete' => $this->delete($input),
+            'restore' => $this->restore($input),
+            'purge' => $this->delete($input, 1),
+            'update' => $this->update($input),
+            'unglobalize' => $this->unglobalize(),
+            default => throw new \RuntimeException(\sprintf("Unsupported object action \"%s\".", $form_action)),
+        };
+    }
```

Comment:
This method is final and therefore cannot be overriden. I think it could directly be placed in the controller itself, as a protected method. The `match` instruction could even be put directly in replacement of the call to this method.